dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,412 @@
+# Crediting the Treasury from existing UTXOs is handled by a new <code>treasury add</code> stake transaction type that burns the UTXO(s) and credits the Treasury account accordingly.  This is analogous to purchasing a ticket.
+# Spending treasury funds now requires stakeholders to vote on the expenditure. The Politeia (or Pi) key operators create a <code>treasury spend</code> transaction and sign it.  This transaction is then broadcast onto the network where it enters the <code>mempool</code>.  Voting wallets are expected to detect this transaction and vote <code>Yes</code> or <code>No</code> on the expenditure based on stakeholder preference via the existing <code>vote</code> stake transactions.  The <code>treasury spend</code> transaction is not valid until enough <code>Yes</code> votes are cast by the stakeholders at which point it is included in a block thereby finalizing the payouts.
+
+==Specification==
+
+The following section details the new transaction types.  All new
+transactions shall be part of the stake tree and all treasury opcodes shall be
+rejected if used in scripts in the regular tree.
+
+===Treasurybase===
+
+The newly introduced treasurybase moves the developer subsidy from the coinbase
+transaction to its own transaction in the stake tree. The treasurybase
+transaction shall be the first transaction in the stake tree.  There shall be
+no more than one treasurybase transaction per block. The developer subsidy
+shall credit the treasury account in full unlike the current implementation
+which credits a proportion based on the number of tickets included in the
+block.
+
+A <code>treasurybase</code> transaction is identified by the following characteristics:
+* The wire message transaction version shall be 3.
+* One TxIn that shall *NOT* have a signature script.
+* All TxOut scripts shall be version 0.
+* There must be two TxOut scripts.
+** TxOut[0] script must be <code>OP_TADD</code>.
+** TxOut[1] shall be an <code>OP_RETURN</code> followed by a twelve byte data push <code>OP_DATA_12</code> and corresponding data for its script.
+* The previous transaction output point shall be null.
+
+{|
+!Transaction!!Script!!Description
+|-
+|TxIn[0]||No script||Stakebase and ValueIn shall be exact development subsidy.
+|-
+|TxOut[0]||[0]OP_TADD||Value shall be exact development subsidy.
+|-
+|TxOut[1]||[0]OP_RETURN<br>[1]OP_DATA_12<br>[2:6][4]{LE encoded height}<br>[6:14][8]{random}||Block height shall be little endian encoded in [2:6] script bytes.
+|}
+
+Additional consensus enforced characteristics:
+
+Note that some rules have been enforced prior the ones listed but out of an
+abundance of caution some tests are repeated in places where it makes sense.
+This is normal and desired.
+
+* First transaction of the stake tree shall be a treasurybase.
+* There shall be only one treasurybase transaction in the stake tree.
+* The previous out point for TxIn[0] shall be null.
```

Comment:
Same deal here regarding all the `TxIn[0]`.  I'll refrain from commenting on the rest, but all should be changed.